### ISSUE TITLE
Error in FireFox 41.0

### ISSUE THREAD

**Issue by ambethoney** [lhorie/mithril.js#817]
function setAttributes(node, tag, dataAttrs, cachedAttrs, namespace) {
		for (var attrName in dataAttrs) {
			var dataAttr = dataAttrs[attrName];
			var cachedAttr = cachedAttrs[attrName];
			if (!(attrName in cachedAttrs) || (cachedAttr !== dataAttr)) {
				cachedAttrs[attrName] = dataAttr;
				try {
					//`config` isn't a real attributes, so ignore it
					if (attrName === "config" || attrName == "key") continue;
					//hook event handlers to the auto-redrawing system
					else if (typeof dataAttr === FUNCTION && attrName.indexOf("on") === 0) {
						node[attrName] = autoredraw(dataAttr, node)
					}
					//handle `style: {...}`
					else if (attrName === "style" && dataAttr != null && type.call(dataAttr) === OBJECT) {
						for (var rule in dataAttr) {
							if (cachedAttr == null || cachedAttr[rule] !== dataAttr[rule]) node.style[rule] = dataAttr[rule]
						}
						for (var rule in cachedAttr) {
							if (!(rule in dataAttr)) node.style[rule] = ""
						}
					}
					//handle SVG
					else if (namespace != null) {
						if (attrName === "href") node.setAttributeNS("http://www.w3.org/1999/xlink", "href", dataAttr);
						else if (attrName === "className") node.setAttribute("class", dataAttr);
						else node.setAttribute(attrName, dataAttr)
					}
					//handle cases that are properties (but ignore cases where we should use setAttribute instead)
					//- list and form are typically used as strings, but are DOM element references in js
					//- when using CSS selectors (e.g. `m("[style='']")`), style is used as a string, but it's an object in js
					else if (attrName in node && !(attrName === "list" || attrName === "style" || attrName === "form" || attrName === "type" || attrName === "width" || attrName === "height")) {
						//#348 don't set the value if not needed otherwise cursor placement breaks in Chrome
>>>		if (tag !== "input" || node[attrName] !== dataAttr) node[attrName] = dataAttr
					}
					else node.setAttribute(attrName, dataAttr)
				}
				catch (e) {
					//swallow IE's invalid argument errors to mimic HTML's fallback-to-doing-nothing-on-invalid-attributes behavior
					if (e.message.indexOf("Invalid argument") < 0) throw e
				}
			}

			else if (attrName === "value" && tag === "input" && node.value != dataAttr) {
				node.value = dataAttr
			}
		}
		return cachedAttrs
	}

Error is ```IndexSizeError: Index or size is negative or greater than the allowed amount```.

The offending line is marked with >>>

**Comment by Zhouzi** [comment]
Providing the code that's generating the error could be really helpful ^^

**Comment by ludbek** [comment]
I am getting similar error in following line
```javascript
	//#348 don't set the value if not needed otherwise cursor placement breaks in Chrome
						if (tag !== "input" || node[attrName] !== dataAttr) node[attrName] = dataAttr
```

**Comment by ludbek** [comment]
So the problem was in following situation. I have a `Textarea` component:
```coffeescript
TextArea = {
  controller: field_controller
  view: (c, pl) ->
    model = pl.model
    field = pl.field

    m '.form-group',
      m 'label.control-label.col-sm-2', {for: field}, pl.placeholder
      m '.col-sm-10',
        m '.bg-danger',
          _.map model.errors[field], (error) ->
            m 'p', error
        m 'textarea.form-control', {
          placeholder: pl.placeholder
          name: field,
          onchange: m.withAttr('value', c.validate),
          value: model.fields[field],
          rows: pl.rows
        }
  }
```

**Comment by bronevichok** [comment]
http://jsfiddle.net/vh3yb6zf/2/ this fiddle shows the problem. FF 44.0.2

**Comment by isiahmeadows** [comment]
Test against 0.2.3, and if that is still the case, try the unminified version in `next`. There were a few bug fixes in 0.2.3 from 0.2.2-rc.

**Comment by lhorie** [comment]
Please re-file with reproduction steps. As it stands, I can't really take any action on this issue
